src/fsharp/NameResolution.fs

```diff
@@ -452,14 +452,16 @@ type BulkAdd = Yes | No
 /// or module, when we collapse the value table down to a dictionary.
 let AddValRefsToItems (bulkAddMode: BulkAdd) (eUnqualifiedItems: LayeredMap<_,_>) (vrefs:ValRef[]) =
     // Object model members are not added to the unqualified name resolution environment 
-    let vrefs = vrefs |> Array.filter (fun vref -> vref.MemberInfo.IsNone)
-
-    if vrefs.Length = 0 then eUnqualifiedItems else
-
     match bulkAddMode with 
-    | BulkAdd.Yes -> 
-        eUnqualifiedItems.AddAndMarkAsCollapsible(vrefs |> Array.map (fun vref -> KeyValuePair(vref.LogicalName, Item.Value vref)))
+    | BulkAdd.Yes ->
+        let mutable eUnqualifiedItems = eUnqualifiedItems 
+        for vref in vrefs do
+            if vref.MemberInfo.IsNone then
+                eUnqualifiedItems <- eUnqualifiedItems.Add (vref.LogicalName, Item.Value vref)
+        eUnqualifiedItems
     | BulkAdd.No -> 
+        let vrefs = vrefs |> Array.filter (fun vref -> vref.MemberInfo.IsNone)
+        if vrefs.Length = 0 then eUnqualifiedItems else
```

Comment:
This to keep the logic like before, but I *think* these two lines make no sense.